src/COLAMD.jl

```diff
@@ -120,3 +120,45 @@ colamd computes a permutation vector `p` such that the Cholesky factorization of
   `A[:,p]' * A[:,p]` has less fill-in and requires fewer floating point operations than `AᵀA`.
 """
 colamd
+
+for (fn, typ) in ((:_symamd, Cint), (:_symamd_l, _Clong))
+
+  @eval begin
+
+    function symamd(A::SparseMatrixCSC{F,$typ}, meta::Colamd{$typ}) where F
+      nrow, ncol = size(A)
+      colptr = A.colptr .- $typ(1)  # 0-based indexing
+      rowval = A.rowval .- $typ(1)
```

Comment:
Isn't it enough to say `.- 1`? The type of `colptr` and `rowval` should match `typ`.